docs/painless/painless-guide/painless-ingest.asciidoc

```diff
@@ -0,0 +1,106 @@
+[[painless-ingest]]
+=== Using Ingest Processors in Painless
+
+Some {ref}/processors.html[ingest processors] expose behavior through Painless
+methods that can be called in Painless scripts that execute in ingest pipelines.
+
+==== Method Usage
+
+All ingest methods available in Painless are scoped to the `Processors`
+namespace. For example:
+
+[source,js]
+--------------------------------------------------
+POST /_ingest/pipeline/_simulate?verbose
+{
+  "pipeline": {
+    "processors": [
+      {
+        "script": {
+          "lang": "painless",
+          "source": """
+            long bytes = Processors.bytes(ctx.size);
+            ctx.size_in_bytes = bytes;
+          """
+        }
+      }
+    ]
+  },
+  "docs": [
+    {
+      "_source": {
+        "size": "1kb"
+      }
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+
+==== Ingest Methods Reference
```

Comment:
I don't love the formatting for these methods. Happy to change to something else if it can be improved.